template/src/components/clicker/index.js

```diff
@@ -13,30 +13,31 @@ function mapStateToProps(state) {
 }
 
 function mapDispatchToProps(dispatch) {
-  return bindActionCreators({ click }, dispatch);
+  return bindActionCreators({ reducer }, dispatch);
 }
 
-@connect(mapStateToProps, mapDispatchToProps)
-export default class Clicker extends React.Component {
+export class Clicker extends React.Component {
   static propTypes = {
     clicker: React.PropTypes.number.isRequired,
-    click: React.PropTypes.func.isRequired,
+    reducer: React.PropTypes.func.isRequired,
+    onClick: React.PropTypes.func.isRequired,
+  }
+  static defaultProps= {
+    onClick: () => {
+      const inputValue = this.refs.incInput.value.trim();
+      this.props.reducer(parseInt(inputValue, 10));
+    },
   }
-
   render() {
     return (
       <div className={styles.main}>
         <span className={styles.clicker}>{ this.props.clicker } +</span>
         <input className={styles.input} ref="incInput" type="text" defaultValue="1" />
-
-        <button onClick={ () => {
-          const inputValue = this.refs.incInput.value.trim();
-          this.props.click(parseInt(inputValue, 10));
-        }}
-        >
+        <button onClick= {this.props.onClick}>
```

Comment:
Made onClick function a property in order to make the component more testable for spies.